airflow/providers/microsoft/azure/hooks/adls.py

```diff
@@ -0,0 +1,264 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from typing import Any
+
+from azure.core.exceptions import ResourceExistsError, ResourceNotFoundError
+from azure.identity import ClientSecretCredential
+from azure.storage.filedatalake import (
+    DataLakeDirectoryClient,
+    DataLakeFileClient,
+    DataLakeServiceClient,
+    DirectoryProperties,
+    FileSystemClient,
+    FileSystemProperties,
+)
+
+from airflow.providers.microsoft.azure.hooks.wasb import WasbHook
+
+
+class AzureDataLakeStorageClient(WasbHook):
```

Comment:
https://github.com/apache/airflow/issues/28223 description states that wasb is legacy if so what is the motivation to inhert from wsab for adsl gen2?

Reply:
Only for connection details, should I create new connection ?

Reply:
If wasb is legacy and we expect it to be removed some day then I don't think we utlize it for other services
but I'm not Azure expert - I leave this desicion to people who knows the service.

What i do want to have here is docs and specifically connection docs that explains how to make this work.
(I want to avoid questions raised down the road by users who will ask to refactor because we utlize legacy service)

Reply:
Ok let me add New connection and with docs

Reply:
@bharanidharan14 @eladkal  did we consider having an `AzureBaseHook` similar to `GoogleBaseHook`
https://github.com/apache/airflow/blob/main/airflow/providers/google/common/hooks/base_google.py
This way, both `AzureDataLakeStorageHook` and `WasbHook` could inherit from it

Reply:
@tatiana and @mateusoliveiraowshq as per our conversation, implementing the `AzureBaseHook` would be challenging since Azure unfortunately does not share the same connection client principles (BlobServiceClient & DataLakeServiceClient) between the WASB and ABFS protocol.

[Azure Blob Storage - WASB ](https://pypi.org/project/azure-storage-blob/)
[Azure Data Lake Storage Gen2 - ABFS](https://pypi.org/project/azure-storage-file-datalake/)

What we could do instead of is to have the following hook structure:

[airflow.providers.microsoft.azure.hooks.wasb](https://airflow.apache.org/docs/apache-airflow-providers-microsoft-azure/stable/_api/airflow/providers/microsoft/azure/hooks/wasb/index.html) (Already Exists)

[airflow.providers.microsoft.azure.hooks.azure_data_lake](https://airflow.apache.org/docs/apache-airflow-providers-microsoft-azure/1.0.0/_api/airflow/providers/microsoft/azure/hooks/azure_data_lake/index.html#module-airflow.providers.microsoft.azure.hooks.azure_data_lake) (Suggested)

Meaning that as per @bharanidharan14 @eladkal conversation we would recommend to inherid the hook from the **airflow.providers.microsoft.azure.hooks.azure_data_lake** instead of **airflow.providers.microsoft.azure.hooks.wasb**, that would make more sense since WASB protocol has been marked as legacy hence the, implementation process are different from each other.



![wasb-abfs](https://user-images.githubusercontent.com/20648427/208137163-382b616f-d14b-413c-afa6-0c7506db95e6.png)

Reply:
@luanmorenomaciel I don't think inheriting from the **airflow.providers.microsoft.azure.hooks.azure_data_lake** is also not recommended because this ADLS existing hook is gen1 and it's being getting retired from feb. They may stop supporting for that too as well. so its better to inherit from `BaseHook`. WDYT ?

Reply:
We should be fine to inherit from BaseHook

Reply:
@luanmorenomaciel Currently I am inheriting from  BaseHook. Can you also review this PR